astropy/cosmology/_src/flrw/base.py

```diff
@@ -244,7 +246,7 @@ def __post_init__(self):
     # Parameter details
 
     @Ob0.validator
-    def Ob0(self, param, value):
+    def Ob0(self, param: Parameter, value: Any) -> float:
```

Comment:
It looks like `Parameter` should be a generic type, but given that Python 3.12 has much better syntax for dealing with that it is not worth implementing while we still support Python 3.11.

Reply:
Agreed.

Reply:
I you guys are worried about the cost of refactoring `TypeVar`s to Python 3.12 syntax, doesn't ruff have auto fixes for it ? https://docs.astral.sh/ruff/rules/non-pep695-type-alias/